server/hotrod/src/main/scala/org/infinispan/server/hotrod/iteration/IterationManager.scala

```diff
@@ -80,7 +80,7 @@ class DefaultIterationManager(val cacheManager: EmbeddedCacheManager) extends It
 
    override def start(cacheName: String, segments: Option[JavaBitSet], filterConverterFactory: Option[String], batch: Integer): IterationId = {
       val iterationId = UUID.randomUUID().toString
-      val stream = cacheManager.getCache(cacheName).getAdvancedCache.cacheEntrySet.stream().asInstanceOf[CacheStream[_]]
+      val stream = cacheManager.getCache(cacheName).getAdvancedCache.cacheEntrySet.stream().asInstanceOf[CacheStream[CacheEntry[AnyRef, AnyRef]]]
```

Comment:
@galderz can you take a look at the typing here?  Both @gustavonalle and I had issues with scala working with Java covariance here as it always wants to return a Stream instead of CacheStream as the CacheCollection defines.